src/lib.rs

```diff
@@ -43,6 +43,11 @@ pub enum Error {
 
     #[error("Unsupported image pull secret type: {0:?}, should be kubernetes.io/dockerconfigjson")]
     BadImagePullSecretType(Option<String>),
+
+    #[error("Finalizer Error: {0}")]
+    // NB: awkward type because finalizer::Error embeds the reconciler error (which is this)
+    // so boxing this error to break cycles
+    FinalizerError(#[source] Box<kube::runtime::finalizer::Error<Error>>),
```

Comment:
Taken directly from the `controller-rs` reference https://github.com/kube-rs/controller-rs/blob/6adbb244c5f35e3d84b4040f824d088e8c9e6bfd/src/lib.rs#L12-L14

I can change this to a specific `kubit` error if we deem it necessary, although I thought that this `FinalizerError` was relatively generic throughout and okay to use.